src/System.Windows.Forms/src/System/Windows/Forms/TaskDialogPage.cs

```diff
@@ -758,26 +761,28 @@ internal void Validate()
                 if (standardButton.IsCreated)
                 {
                     buttonFlags |= standardButton.GetStandardButtonFlag();
-
-                    if (standardButton.DefaultButton && defaultButtonID == 0)
-                    {
-                        defaultButtonID = standardButton.ButtonID;
-                    }
                 }
             }
 
             for (int i = 0; i < _boundCustomButtons.Length; i++)
             {
                 TaskDialogButton customButton = _boundCustomButtons[i];
                 flags |= customButton.Bind(this, CustomButtonStartID + i);
+            }
 
-                if (customButton.IsCreated)
+            if (DefaultButton != null)
+            {
+                // ensure the default button is part of the button collection
+                if (!buttons.Contains(DefaultButton))
                 {
-                    if (customButton.DefaultButton && defaultButtonID == 0)
-                    {
-                        defaultButtonID = customButton.ButtonID;
-                    }
+                    throw new InvalidOperationException(SR.TaskDialogDefaultButtonMustExistInCollection);
                 }
+
+                defaultButtonID = DefaultButton.ButtonID;
+            }
+            else if (DefaultButton == null && _buttons.Any())
+            {
+                defaultButtonID = _buttons.First().ButtonID;
             }
```

Comment:
The `defaultButtonID` can stay `0` when no default button was specified, in which case the first button will automatically be the default one.
```suggestion
```

Reply:
Thank you